libraries/Microsoft.Bot.Connector/Authentication/MicrosoftGovernmentAppCredentials.cs

```diff
@@ -62,5 +65,34 @@ public override string OAuthEndpoint
         {
             get { return GovernmentAuthenticationConstants.ToChannelFromBotLoginUrl; }
         }
+
+        protected override Lazy<IAuthenticator> BuildIAuthenticator()
+        {
+            return new Lazy<IAuthenticator>(
+                () =>
+                {
+                    var credential = new ClientCredential(MicrosoftAppId, MicrosoftAppPassword);
+
+                    IAuthenticator Make(string authority, string scope)
+                        => new AdalAuthenticator(
+                            credential,
+                            new OAuthConfiguration() { Authority = authority, Scope = scope },
+                            this.CustomHttpClient,
+                            this.Logger);
+
+                    if (OAuthScope == GovernmentAuthenticationConstants.ToChannelFromBotOAuthScope)
+                    {
+                        var optionV1 = Make(GovernmentAuthenticationConstants.ToChannelFromBotLoginUrl, GovernmentAuthenticationConstants.ToChannelFromBotOAuthScope);
+                        var optionV2 = Make(GovernmentAuthenticationConstants.ToChannelFromBotLoginUrlV2, GovernmentAuthenticationConstants.ToChannelFromBotOAuthScopeV2);
+
+                        return new Authenticators(optionV2, optionV1);
+                    }
+                    else
+                    {
+                        return Make(OAuthEndpoint, OAuthScope);
```

Comment:
I don't get how this branch of the "IF" statement returns an "Make()" function result, while the other returns a new set of authenticators. 

... yet the parent function needs the lazy.

There's got to be a more clear way to express this to simple minds like mine.

Reply:
@carlosscastro can you give some insight into the need for the two methods (with Lazy<AdalAuthenticator> and Lazy<IAuthenticator>)?  I tried to copy the existing code pattern.